app/lib/features/chat_ng/providers/chat_room_messages_provider.dart

```diff
@@ -49,3 +53,39 @@ final renderableChatMessagesProvider =
     return _supportedTypes.contains(msg.eventItem()?.eventType());
   }).toList();
 });
+
+final mentionSuggestionsProvider =
+    StateProvider.family<Map<String, String>?, (String, MentionType)>(
+        (ref, params) {
+  final roomId = params.$1;
+  final mentionType = params.$2;
+  final client = ref.watch(alwaysClientProvider);
+  final userId = client.userId().toString();
+
+  switch (mentionType) {
+    case MentionType.user:
+      final members = ref.watch(membersIdsProvider(roomId)).valueOrNull;
+      if (members != null) {
+        return members.fold<Map<String, String>>({}, (map, uId) {
+          if (uId != userId) {
+            final displayName = ref.watch(
+              memberDisplayNameProvider(
+                (roomId: roomId, userId: uId),
+              ),
+            );
+            map[uId] = displayName.valueOrNull ?? '';
+          }
+          return map;
+        });
+      }
+
+    case MentionType.room:
+      final rooms = ref.watch(chatIdsProvider);
+      return rooms.fold<Map<String, String>>({}, (map, roomId) {
+        final displayName = ref.watch(roomDisplayNameProvider(roomId));
+        map[roomId] = displayName.valueOrNull ?? '';
+        return map;
+      });
+  }
+  return null;
+});
```

Comment:
I was a bit hesitant to write too many providers while it can be combined just in one. But that's just my personal preference. If it improves understanding, then I'm happy to split :)